port/cpl_multiproc.cpp

```diff
@@ -1462,35 +1462,31 @@ int CPLCreateOrAcquireMutexEx(CPLMutex **phMutex, double dfWaitInSeconds,
                               int nOptions)
 
 {
-    bool bSuccess = false;
-
     pthread_mutex_lock(&global_mutex);
     if (*phMutex == nullptr)
     {
         *phMutex = CPLCreateMutexInternal(true, nOptions);
-        bSuccess = *phMutex != nullptr;
+        const bool bSuccess = *phMutex != nullptr;
```

Comment:
are the changes in this file related to the TSAN complaint mentioned above?

On a side note, is there chance we can eventually get rid of in-house mutex handling in favor of std:: library methods?

Reply:
> are the changes in this file related to the TSAN complaint mentioned above?

not directly, I also incorporated that commit from #1108 because a sanitizer that would test this PR would also frown upon that issue

> On a side note, is there chance we can eventually get rid of in-house mutex handling in favor of std:: library methods?

could be a nice cleanup indeed.